draco/analysis/flagging.py

```diff
@@ -1134,50 +1135,74 @@ class MaskFreq(task.SingleTask):
         start and end indices of a range to flag (as with a standard slice, the end
         is *not* included.)
     factorize : bool, optional
-        Find the smallest factorizable mask of the time-frequency axis that covers all samples already flagged in the data.
+        Find the smallest factorizable mask of the time-frequency axis that covers all
+        samples already flagged in the data.
     all_time : bool, optional
         Only include frequencies where all time samples are present.
-    mask_missing_baselines : bool, optional
+    mask_missing_data : bool, optional
         Mask time-freq samples where some baselines are missing.
     """
 
     bad_freq_ind = config.Property(proptype=list, default=None)
     factorize = config.Property(proptype=bool, default=False)
     all_time = config.Property(proptype=bool, default=False)
-    mask_missing_baselines = config.Property(proptype=bool, default=False)
+    mask_missing_data = config.Property(proptype=bool, default=False)
 
-    def process(self, data):
+    def process(
+        self, data: Union[containers.VisContainer, containers.RingMap]
+    ) -> Union[containers.RFIMask, containers.SiderealRFIMask]:
         """Apply the mask to the data.
 
         Parameters
         ----------
-        data : SiderealStream or TimeStream
+        data
             The data to mask.
 
         Returns
         -------
-        data_masked : SiderealRFIMask or RFIMask
+        data_masked
             The mask marking bad data.
         """
 
         data.redistribute("freq")
 
+        maskcls = (
+            containers.SiderealRFIMask
+            if isinstance(data, containers.SiderealContainer)
+            else containers.RFIMask
+        )
+        maskcont = maskcls(axes_from=data, attrs_from=data)
+        mask = maskcont.mask[:]
+
         # Get the total number of baselines observed at each freq-time. This is used to
         # create an initial mask
-        present_baselines = mpiarray.MPIArray.wrap(
-            (data.weight[:] > 0).sum(axis=1), comm=data.weight.comm, axis=0
-        )
-        all_present_baselines = present_baselines.allgather()
-        mask = all_present_baselines == 0
+        if isinstance(data, containers.VisContainer):
+            present_data = mpiarray.MPIArray.wrap(
+                (data.weight[:] > 0).sum(axis=1), comm=data.weight.comm, axis=0
+            )
+        elif isinstance(data, containers.RingMap):
```

Comment:
Add comment about ringmap case:
```suggestion
        # For ringmap, get total number of elevations/pols observed at each freq and RA
        elif isinstance(data, containers.RingMap):
```